Database/CleanArchive.py

```diff
@@ -0,0 +1,99 @@
+                updates['datetime'] += 1
+            except:
+                print('failed parsing date from date')
+                failures['datetime'] += 1
+        if 'metadata' in episode:
+            # Parse datetime from Subtitle attrubite. Includes hour, endtime, and timezone.
+            # Stores datetime in UTC time.
+            try:
+                subtitle = episode['metadata']['Subtitle']
+                # Match 'CSPAN July 16, 2009 11:00pm-2:00am EDT'
+                time = '(\d+):(\d+)(\w+)'
+                match = re.match(f'.* (\w+) (\d+), (\d+) {time}-{time} (\w+)', subtitle)
+                month, day, year, hour, minute, ampm, hour_end, minute_end, ampm_end, timezone = match.groups()
+                date = datetime.strptime(f'{year} {month} {day} {hour}:{minute} {ampm.upper()}', '%Y %B %d %I:%M %p')
+                if timezone in ['EST', 'EDT']:
+                    timezone = 'EST5EDT'
+                elif timezone in ['PST', 'PDT']:
+                    timezone = 'PST8PDT'
+                elif timezone in ['CST', 'CDT']:
+                    timezone = 'CST6CDT'
+                date = pytz.timezone(timezone).localize(date)
+                set_fields['metadata.Datetime_UTC'] = date.astimezone(pytz.utc)
+                updates['metadata.Subtitle'] += 1
+            except:
+                print('Failed parsing date from subtitle.')
+                failures['metadata.Subtitle'] += 1
+
+            # Parse duration into timedelta object. Convert to seconds and store.
+            try:
+                duration = datetime.strptime(episode['metadata']['Duration'], '%H:%M:%S')
+                duration_delta = timedelta(hours=duration.hour, minutes=duration.minute, seconds=duration.second)
+                set_fields['metadata.Duration_s'] = duration_delta.seconds
+                if duration_delta.seconds == 0:
+                    errors.append('duration_is_0')
+                updates['metadata.Duration'] += 1
+            except:
+                print('Failed parsing timedelta from duration')
+                failures['metadata.Duration'] += 1
+
+            # Mark episodes as empty or short according to text length.
+            try:
+                transcript_len = 0
+                for snippet in episode['snippets']:
+                    transcript_len += len(snippet['transcript'])
+                set_fields['transcript_str_length'] = transcript_len
+                if transcript_len == 0:
+                    errors.append('transcript_is_empty')
+                if transcript_len < 244:
+                    errors.append('transcript_is_short')
+                updates['transcript_len'] += 1
+            except:
+                print('Failed calculating segment length')
+                failures['transcript_len'] += 1
+
+        # Remove empty transcripts.
+        transaction = {
+            '$pull': {'snippets': {'transcript': ''}}
```

Comment:
should we just remove the whole entry if the transcripts are empty? Does having just the title and other fields benefit us?